crates/net/network/src/transactions.rs

```diff
@@ -1016,6 +1005,95 @@ struct Peer {
     client_version: Arc<String>,
 }
 
+// === TransactionFetcher ===
+/// * active requests - especially inflight hashes, so that whenever we receive new missing hashes
+///   we can check first if they're already being requested.
+/// * missing hashes and peers that send us these - so we can possibly re-request them
+
+#[derive(Debug, Default)]
+struct TransactionFetcher {
+    /// All currently active requests for pooled transactions.
+    inflight_requests: FuturesUnordered<GetPooledTxRequestFut>,
+    /// Set that tracks all hashes that are currently being fetched.
+    inflight_hash_to_fallback_peers: HashMap<TxHash, Vec<PeerId>>,
+}
+
+impl TransactionFetcher {
+    fn poll(
+        &mut self,
+        cx: &mut Context<'_>,
+    ) -> (PeerId, Option<Vec<PooledTransactionsElement>>, Option<RequestError>) {
```

Comment:
for this we want a new enum event type, because I believe we'll have more variants in the future, e.g. `enum FetchEvent`

this also needs to return `Poll<FetchEvent>`, if nothing to do, then this returns pending

Reply:
nice! that makes a ton of sense